### ISSUE TITLE
[Bug] Urgent - Embedding vector no longer getting generated in Azure AI Search using ServerLess mode

### ISSUE THREAD

**Issue by bancroftway** [microsoft/kernel-memory#806]
### Context / Scenario

On my older projects involving Kernel Memory, I always set EmbeddingDimensions property of AzureOpenAIConfig to 1536. Recently I started getting an error "This model does not support specifying dimensions". I opened this [issue ](https://github.com/microsoft/kernel-memory/issues/805) and as per the suggestion in there, I set this property to null. Now when I ingest pdf files, I do not see embedding vector in Azure AI Search Index as before. The AskAsync is not working either.

In the payload field, towards the end, I just see this text :
`\"vector_provider\":\"AI.AzureOpenAI.AzureOpenAITextEmbeddingGenerator\",\"vector_generator\":\"TODO\",\"last_update\":\"2024-09-28T11:23:02\"`

![image](https://github.com/user-attachments/assets/68151c99-2a1f-42f2-9135-6f1e4cf5329e)

Here is my code:

```
public async Task<string> AskQuestionFromDocuments(string question, Dictionary<string, string> tagDictionary)
{
    var skKernel = GetSkKernel();
    var kmKernel = GetKernelMemoryClient();

    var skPrompt = """
    Based only on the facts recalled from memory in triple backquotes, answer this question. If the answer is not known
    say 'I don't know' otherwise generate a clear, detailed answer suitable for 
    a novice to the topic at hand. Prefix with one emoji relevant to the content.
    Important: If the facts from memory are blank or are not enough to answer the question, do not answer the question, and respond with 'I don't know'.
    Question: {{$input}}

    Facts recalled from memory: 
    ```
    {{$facts}}
    ```
    """
    ;

    MemoryFilter memoryFilter = new MemoryFilter();
    if (tagDictionary?.Any() == true)
    {
        foreach (var kvp in tagDictionary)
        {
            memoryFilter = memoryFilter.ByTag(kvp.Key, kvp.Value);
        }
    }            

    var factsFromMemory = await kmKernel.AskAsync(question, filter: memoryFilter);

    OpenAIPromptExecutionSettings settings = new()
    {
        Temperature = 0.70,
    };

    var context = new KernelArguments(settings)
    {
        ["input"] = question,
        ["facts"] = factsFromMemory.Result
    };

    var myFunction = skKernel.CreateFunctionFromPrompt(skPrompt);

    var answer = await myFunction.InvokeAsync(skKernel, context);

    return answer.GetValue<string>() ?? string.Empty;
}

private MemoryServerless GetKernelMemoryClient()
{

    var azureOpenAITextConfig = new AzureOpenAIConfig
    {
        APIType = AzureOpenAIConfig.APITypes.ChatCompletion,
        MaxRetries = 10,
        MaxTokenTotal = configuration.GetValue<int>("AzureServices:OpenAI:LlmMaxTokens"),
        Endpoint = configuration["AzureServices:OpenAI:Url"]!,
        APIKey = configuration["AzureServices:OpenAI:ApiKey"]!,
        Auth = AzureOpenAIConfig.AuthTypes.APIKey,
        Deployment = configuration["AzureServices:OpenAI:ModelName"]!
    };

    var azureOpenAIEmbeddingConfig = new AzureOpenAIConfig()
    {
        APIType = AzureOpenAIConfig.APITypes.EmbeddingGeneration,
        MaxRetries = 10,
        MaxTokenTotal = configuration.GetValue<int>("AzureServices:OpenAI:EmbeddingMaxTokens"),
        Endpoint = configuration["AzureServices:OpenAI:Url"]!,
        APIKey = configuration["AzureServices:OpenAI:ApiKey"]!,
        Auth = AzureOpenAIConfig.AuthTypes.APIKey,
        Deployment = configuration["AzureServices:OpenAI:EmbeddingModel"]!,
        EmbeddingDimensions = null,
        MaxEmbeddingBatchSize = configuration.GetValue<int>("AzureServices:OpenAI:MaxEmbeddingBatchSize"),
    };

    var azureBlobsConfig = new AzureBlobsConfig()
    {
        Auth = AzureBlobsConfig.AuthTypes.ConnectionString,
        ConnectionString = configuration["ConnectionStrings:StorageConnection"]!,
        Container = configuration["AzureServices:OpenAI:EmbeddingsStorageContainer"]!
    };

    var azureAISearchConfig = new AzureAISearchConfig()
    {
        Auth = AzureAISearchConfig.AuthTypes.APIKey,
        APIKey = configuration["AzureServices:Search:AdminApiKey"]!,
        Endpoint = configuration["AzureServices:Search:EndPoint"]!,
        UseHybridSearch = true,
    };

    var memory = new KernelMemoryBuilder()
        .WithAzureOpenAITextGeneration(azureOpenAITextConfig)
        .WithAzureOpenAITextEmbeddingGeneration(azureOpenAIEmbeddingConfig)
        .WithAzureBlobsDocumentStorage(azureBlobsConfig)
        .WithAzureAISearchMemoryDb(azureAISearchConfig)
        .WithSearchClientConfig(new SearchClientConfig { MaxMatchesCount = 4, Temperature = 0, TopP = 0 })
        .Build<MemoryServerless>();

    return memory;
}

private Kernel GetSkKernel()
{
    var skKernel = Kernel.CreateBuilder()
    .AddAzureOpenAIChatCompletion(
        configuration["AzureServices:OpenAI:ModelName"]!,
        configuration["AzureServices:OpenAI:Url"]!,
        apiKey: configuration["AzureServices:OpenAI:ApiKey"]!
    )
    .Build();

    return skKernel;
}
```

### What happened?

The embedding vector is no longer present in Azure AI search

### Importance

edge case

### Platform, Language, Versions

.Net 9 RC1

**Comment by bancroftway** [comment]
Using the above code, I ingested 'Alice in Wonderland' text file (attached) to eliminate any issues with pdfs and Office files. The embedding vector is not getting generated. Here is the first snippet of the json search results:

```
{
  "@odata.context": "https://search-whisperfyi-dev.search.windows.net/indexes('default')/$metadata#docs(*)",
  "@odata.count": 52,
  "@search.nextPageParameters": {
    "search": "*",
    "count": true,
    "skip": 50
  },
  "value": [
    {
      "@search.score": 1,
      "id": "ZD02Ly9wPWQxZWNmNTIyYjBiYzRhNTRhNTkwZGJjN2M3YzU3MmQ4",
      "tags": [
        "__document_id:6",
        "__file_type:text/plain",
        "__file_id:679d5fc7430c4451857d436b7c019148",
        "__file_part:f267720ad82b4018b918dbc873c889fc",
        "__part_n:2",
        "__sect_n:0",
        "productid:1",
        "categoryid:6"
      ],
      "payload": "{\"url\":\"\",\"schema\":\"20231218A\",\"file\":\"AliceInWonderLand.txt\",\"text\":\"the fall was over.\\n\\nAlice was not a bit hurt, and she jumped up on to her feet in a moment:\\nshe looked up, but it was all dark overhead; before her was another\\nlong passage, and the White Rabbit was still in sight, hurrying down\\nit. There was not a moment to be lost: away went Alice like the wind,\\nand was just in time to hear it say, as it turned a corner, \\uFFFDOh my ears\\nand whiskers, how late it\\uFFFDs getting!\\uFFFD She was close behind it when she\\nturned the corner, but the Rabbit was no longer to be seen: she found\\nherself in a long, low hall, which was lit up by a row of lamps hanging\\nfrom the roof.\\r\\nThere were doors all round the hall, but they were all locked; and when\\nAlice had been all the way down one side and up the other, trying every\\ndoor, she walked sadly down the middle, wondering how she was ever to\\nget out again.\\n\\nSuddenly she came upon a little three-legged table, all made of solid\\nglass; there was nothing on it except a tiny golden key, and Alice\\uFFFDs\\nfirst thought was that it might belong to one of the doors of the hall;\\nbut, alas! either the locks were too large, or the key was too small,\\nbut at any rate it would not open any of them. However, on the second\\ntime round, she came upon a low curtain she had not noticed before, and\\r\\nbehind it was a little door about fifteen inches high: she tried the\\nlittle golden key in the lock, and to her great delight it fitted!\\n\\nAlice opened the door and found that it led into a small passage, not\\nmuch larger than a rat-hole: she knelt down and looked along the\\npassage into the loveliest garden you ever saw. How she longed to get\\nout of that dark hall, and wander about among those beds of bright\\nflowers and those cool fountains, but she could not even get her head\\nthrough the doorway; \\uFFFDand even if my head would go through,\\uFFFD thought\\npoor Alice, \\uFFFDit would be of very little use without my shoulders. Oh,\\r\\nhow I wish I could shut up like a telescope! I think I could, if I only\\nknew how to begin.\\uFFFD For, you see, so many out-of-the-way things had\\nhappened lately, that Alice had begun to think that very few things\\nindeed were really impossible.\\n\\nThere seemed to be no use in waiting by the little door, so she went\\nback to the table, half hoping she might find another key on it, or at\\nany rate a book of rules for shutting people up like telescopes: this\\ntime she found a little bottle on it, (\\uFFFDwhich certainly was not here\\nbefore,\\uFFFD said Alice,) and round the neck of the bottle was a paper\\nlabel, with the words \\uFFFDDRINK ME,\\uFFFD beautifully printed on it in large\\r\\nletters.\\n\\nIt was all very well to say \\uFFFDDrink me,\\uFFFD but the wise little Alice was\\nnot going to do _that_ in a hurry. \\uFFFDNo, I\\uFFFDll look first,\\uFFFD she said,\\n\\uFFFDand see whether it\\uFFFDs marked \\uFFFD_poison_\\uFFFD or not\\uFFFD; for she had read\\nseveral nice little histories about children who had got burnt, and\\neaten up by wild beasts and other unpleasant things, all because they\\n_would_ not remember the simple rules their friends had taught them:\\nsuch as, that a red-hot poker will burn you if you hold it too long;\\nand that if you cut your finger _very_ deeply with a knife, it usually\\nbleeds; and she had never forgotten that, if you drink much from a bottle marked \\uFFFDpoison,\\uFFFD it is almost certain to disagree with you,\\nsooner or later.\\n\\nHowever, this bottle was _not_ marked \\uFFFDpoison,\\uFFFD so Alice ventured to\\ntaste it, and finding it very nice, (it had, in fact, a sort of mixed\\n\",\"vector_provider\":\"AI.AzureOpenAI.AzureOpenAITextEmbeddingGenerator\",\"vector_generator\":\"TODO\",\"last_update\":\"2024-09-28T11:47:40\"}"
    },
    {
      "@search.score": 1,
      "id": "ZD02Ly9wPTlhZDc3YjdjOGJkYzQwMDE4MzIwYzc4NDE1MWE2NjA5",
      "tags": [
        "__document_id:6",
        "__file_type:text/plain",
        "__file_id:679d5fc7430c4451857d436b7c019148",
        "__file_part:944932455bbc4552ba364d3aab613975",
        "__part_n:14",
        "__sect_n:0",
        "productid:1",
        "categoryid:6"
      ],
      "payload": "{\"url\":\"\",\"schema\":\"20231218A\",\"file\":\"AliceInWonderLand.txt\",\"text\":\"held it out to the puppy; whereupon the puppy jumped into the air off\\nall its feet at once, with a yelp of delight, and rushed at the stick,\\nand made believe to worry it; then Alice dodged behind a great thistle,\\nto keep herself from being run over; and the moment she appeared on the\\nother side, the puppy made another rush at the stick, and tumbled head\\nover heels in its hurry to get hold of it; then Alice, thinking it was\\nvery like having a game of play with a cart-horse, and expecting every\\nmoment to be trampled under its feet, ran round the thistle again; then\\nthe puppy began a series of short charges at the stick, running a very\\r\\nlittle way forwards each time and a long way back, and barking hoarsely\\nall the while, till at last it sat down a good way off, panting, with\\nits tongue hanging out of its mouth, and its great eyes half shut.\\n\\nThis seemed to Alice a good opportunity for making her escape; so she\\nset off at once, and ran till she was quite tired and out of breath,\\nand till the puppy\\uFFFDs bark sounded quite faint in the distance.\\n\\n\\uFFFDAnd yet what a dear little puppy it was!\\uFFFD said Alice, as she leant\\nagainst a buttercup to rest herself, and fanned herself with one of the\\nleaves: \\uFFFDI should have liked teaching it tricks very much, if\\uFFFDif I\\uFFFDd\\r\\nonly been the right size to do it! Oh dear! I\\uFFFDd nearly forgotten that\\nI\\uFFFDve got to grow up again! Let me see\\uFFFDhow _is_ it to be managed? I\\nsuppose I ought to eat or drink something or other; but the great\\nquestion is, what?\\uFFFD\\n\\nThe great question certainly was, what? Alice looked all round her at\\nthe flowers and the blades of grass, but she did not see anything that\\nlooked like the right thing to eat or drink under the circumstances.\\nThere was a large mushroom growing near her, about the same height as\\nherself; and when she had looked under it, and on both sides of it, and\\nbehind it, it occurred to her that she might as well look and see what\\r\\nwas on the top of it.\\n\\nShe stretched herself up on tiptoe, and peeped over the edge of the\\nmushroom, and her eyes immediately met those of a large blue\\ncaterpillar, that was sitting on the top with its arms folded, quietly\\nsmoking a long hookah, and taking not the smallest notice of her or of\\nanything else.\\n\\n\\n\\n\\nCHAPTER V.\\nAdvice from a Caterpillar\\n\\n\\nThe Caterpillar and Alice looked at each other for some time in\\nsilence: at last the Caterpillar took the hookah out of its mouth, and\\naddressed her in a languid, sleepy voice.\\n\\n\\uFFFDWho are _you?_\\uFFFD said the Caterpillar.\\n\\nThis was not an encouraging opening for a conversation. Alice replied,\\r\\nrather shyly, \\uFFFDI\\uFFFDI hardly know, sir, just at present\\uFFFDat least I know\\nwho I _was_ when I got up this morning, but I think I must have been\\nchanged several times since then.\\uFFFD\\n\\n\\uFFFDWhat do you mean by that?\\uFFFD said the Caterpillar sternly. \\uFFFDExplain\\nyourself!\\uFFFD\\n\\n\\uFFFDI can\\uFFFDt explain _myself_, I\\uFFFDm afraid, sir,\\uFFFD said Alice, \\uFFFDbecause I\\uFFFDm\\nnot myself, you see.\\uFFFD\\n\\n\\uFFFDI don\\uFFFDt see,\\uFFFD said the Caterpillar.\\n\\n\\uFFFDI\\uFFFDm afraid I can\\uFFFDt put it more clearly,\\uFFFD Alice replied very politely,\\n\\uFFFDfor I can\\uFFFDt understand it myself to begin with; and being so many\\ndifferent sizes in a day is very confusing.\\uFFFD\\n\\n\\uFFFDIt isn\\uFFFDt,\\uFFFD said the Caterpillar. \\uFFFDWell, perhaps you haven\\uFFFDt found it so yet,\\uFFFD said Alice; \\uFFFDbut when you\\nhave to turn into a chrysalis\\uFFFDyou will some day, you know\\uFFFDand then\\nafter that into a butterfly, I should think you\\uFFFDll feel it a little\\n\",\"vector_provider\":\"AI.AzureOpenAI.AzureOpenAITextEmbeddingGenerator\",\"vector_generator\":\"TODO\",\"last_update\":\"2024-09-28T11:47:46\"}"
    },
```

**Comment by bancroftway** [comment]
So this was my bad. The embedding vector is being generated, but not being returned in the Azure AI search results. 

It looks like when the Index is created by KM, the embedding_vector field is not being marked as 'Retrievable' by default?

![image](https://github.com/user-attachments/assets/3a690452-c5e8-4bca-8832-8a4e188011f9)

**Comment by bancroftway** [comment]
I will open a new issue regarding the retrievability of the embedding field
